features/mbedtls/targets/TARGET_NUVOTON/TARGET_M480/aes/aes_alt.c

```diff
@@ -21,298 +21,213 @@
  *  http://csrc.nist.gov/publications/fips/fips197/fips-197.pdf
  */
 
-#if !defined(MBEDTLS_CONFIG_FILE)
-#include "mbedtls/config.h"
-#else
-#include MBEDTLS_CONFIG_FILE
-#endif
+#include "mbedtls/aes.h"
 
 #if defined(MBEDTLS_AES_C)
 #if defined(MBEDTLS_AES_ALT)
 
 #include <string.h>
-
-#include "mbedtls/aes.h"
+#include <stdbool.h>
 
 #include "M480.h"
 #include "mbed_toolchain.h"
 #include "mbed_assert.h"
-
-
-
-#define mbedtls_trace(...) //printf(__VA_ARGS__)
+#include "mbed_error.h"
+#include "nu_bitutil.h"
+#include "crypto-misc.h"
 
 /* Implementation that should never be optimized out by the compiler */
-static void mbedtls_zeroize( void *v, size_t n ) {
-    volatile unsigned char *p = (unsigned char*)v; while( n-- ) *p++ = 0;
+static void mbedtls_zeroize( void *v, size_t n )
+{
+    volatile unsigned char *p = (unsigned char*)v;
+    while( n-- ) *p++ = 0;
 }
 
-
-static uint32_t au32MyAESIV[4] = {
-    0x00000000, 0x00000000, 0x00000000, 0x00000000
-};
-
-extern volatile int  g_AES_done;
-
-// Must be a multiple of 16 bytes block size 
+/* DMA compatible backup buffer if user buffer doesn't meet requirements
+ * 
+ * MAX_DMA_CHAIN_SIZE must be a multiple of 16-byte block size.
+ * Its value is estimated to trade memory footprint off against performance. 
+ */
```

Comment:
Note that there are some requirements on the placement of these buffers in the address range so that the DMA can access them. It would be good to:
* Document this clearly here by expanding the comment above
* Check the linker files to ensure that global variables are placed in the accepted address range

Reply:
@andresag01 I refine DMA buffer requirement comment in https://github.com/ARMmbed/mbed-os/pull/4925/commits/b6d4371041e165f9efb19085487eec6369b73a8e.